lib/filterx/filterx-grammar.ym

```diff
@@ -238,44 +237,46 @@ template
           }
 	;
 
-dict_expr
-	: '{' dict_key_values '}'               { $$ = filterx_dict_expr_new(last_fillable, $2); }
+dict_generator
+	: '{' { last_literal_generator = filterx_literal_generator_new(); } dict_elements '}'
+						{ filterx_literal_generator_set_elements(last_literal_generator, $3); $$ = last_literal_generator; }
```

Comment:
yeah, this is where we use last_literal_generator after potentially recursing back to the same rule, which means
that the global variable will be clobbered by the time we return from the first iteration.

Reply:
I think this does not affect us, as we can only have one literal generator, the inner ones will be inner_..._generators.
In other words: You cannot reach dict_generator from another dict_generator.

But I understand that it is not very clean and we might forget about this if we introduce changes later.

Reply:
ok. I've tried to fix this, but got a lot of conflicts, so let's leave it as is.